spec/components/selection/Switch.spec.tsx

```diff
@@ -0,0 +1,86 @@
+import { mount, shallow } from 'enzyme';
+import * as React from 'react';
+import { Switch } from '../../../src/components';
+import SelectionTypes from '../../../src/components/selection/SelectionTypes';
+import { SelectionInput } from '../../../src/components/selection/SelectionInput';
+import SelectionStatus from '../../../src/components/selection/SelectionStatus';
+
+describe('Switch Component', () => {
+  describe('Switch component test suite => ', () => {
+    afterEach(() => {
+      jest.restoreAllMocks();
+    });
+
+    it('render with default props and initial value', () => {
+      const wrapper = shallow(
+        <Switch name="default-checkbox-name" value="default-checkbox-value" />
+      );
+      expect(wrapper.length).toEqual(1);
+      expect(wrapper.find(SelectionInput).length).toBe(1);
+      expect(wrapper.find(SelectionInput).prop('type')).toEqual(
+        SelectionTypes.SWITCH
+      );
+      expect(wrapper.find(SelectionInput).prop('name')).toEqual(
+        'default-checkbox-name'
+      );
+      expect(wrapper.find(SelectionInput).prop('value')).toEqual(
+        'default-checkbox-value'
+      );
+    });
+
+    it('with "checked" status', () => {
+      const onChangeCallback = jest.fn();
+      const wrapper = mount(
+        <Switch
+          name="checked-state-checkbox-name"
+          value="checked-state-checkbox-value"
+          status={SelectionStatus.CHECKED}
+          onChange={onChangeCallback}
+        />
+      );
+      expect(wrapper.length).toEqual(1);
+      expect(wrapper.find(SelectionInput).length).toBe(1);
+      expect(wrapper.find(SelectionInput).prop('type')).toEqual(
+        SelectionTypes.SWITCH
+      );
+      expect(wrapper.find(SelectionInput).prop('status')).toBe(
+        SelectionStatus.CHECKED
+      );
+      wrapper.unmount();
+    });
+
+    it('with "mixed" status', () => {
```

Comment:
You can removed this test because it's for the 'mixed' state of a checkbox. (https://www.w3.org/TR/wai-aria-practices-1.1/examples/checkbox/checkbox-2/checkbox-2.html)
Not applicable for the Switch ;)